### PR TITLE
Add basic support for vector labels

### PR DESCRIPTION
**Pull request by chrismayer**
This adds basic support to configure labels in the style for vector based layers. Base for the label text is a single attribute in the
underlying vector data. A configuration like this can be added to the style section of each vector based layer configuration:

```json
        "label": {
          "attribute": "name",
          "minResolution": 4.0,
          "outlineColor": "white",
          "outlineWidth": 2,
          "fillColor": "blue",
          "offsetX": 0,
          "offsetY": 15,
          "align": "center"
        }
```
Result:
image

Besides `attribute`, `outlineColor`, `outlineWidth`, `fillColor`, `minResolution`, `maxResolution` every  config property of [ol/style/Text](https://openlayers.org/en/latest/apidoc/module-ol_style_Text-Text.html) can be set, e.g. `offsetX` and `offsetY`.

More sophisticated templating can be discussed in #157.

### CODE REVIEW

File: src/factory/OlStyle.js [hunk 1]
```diff
@@ -109,6 +120,56 @@ export const OlStyleFactory = {
...
+  },
+
+  /**
+   * Creates and returns a basic OL style object for texts.
+   *
+   * @param {Object} labelConf Style config object for labels
+   * @returns {ol/style/Text} The OL style object for texts
+   */
+  getTextStyle (labelConf) {
+    const textConf = { ...labelConf };
```
Comment (JakobMiksch) [inline src/factory/OlStyle.js:132]: why is this necessary? 
a comment might be helpful

File: src/factory/OlStyle.js [hunk 2]
```diff
@@ -109,6 +120,56 @@ export const OlStyleFactory = {
...
+   * @param {Object} labelStyleConf
+   * @returns {ol/style/Style~StyleFunction}
+   *    Style function returning the OL style object enriched by texts / labels
+   */
+  getTextStyleFunction (style, labelStyleConf) {
+    const labelStyle = OlStyleFactory.getTextStyle(labelStyleConf);
+    const labelAttr = labelStyleConf.attribute;
+    return (feature, resolution) => {
+      // detect min/max resolution to show labels
+      // if nothing is confured labels are shown regardless of resolution
```
Comment (JakobMiksch) [inline src/factory/OlStyle.js:159]: `confured` - what is meant here?

**Review by JakobMiksch — COMMENTED** [review]
@chrismayer I tried it also with another layer in it works fine. 
Just two minor remarks regarding comments

File: src/factory/OlStyle.js (hunk 1 above)
Comment (chrismayer) [reply src/factory/OlStyle.js:132]: It ensures a clone of the config is created, so no in place modification of the config is done. Good point, I'll add a comment.

File: src/factory/OlStyle.js (hunk 2 above)
Comment (chrismayer) [reply src/factory/OlStyle.js:159]: `configured` , I'll address that.

File: src/factory/OlStyle.js (hunk 1 above)
Comment (chrismayer) [reply src/factory/OlStyle.js:133]: See e32d245

**Comment by JakobMiksch** [comment]
Looks good. Ready to merge.

**Comment by chrismayer** [comment]
Thanks again @JakobMiksch
